app/controllers/application_controller.rb

```diff
@@ -8,10 +8,29 @@ class ApplicationController < ActionController::Base
     around_action :with_highlight_context
   end
   before_action :set_account
+  before_action :setup_installation if Installation.installation_flow?
 
   private
 
+  def setup_installation
+    installation_routes_regex = %r{\A/installation/(new|create|step_1|step_2|step_3|update_step_1|update_step_2|update_step_3|loading)\z}
+    return unless Installation.installation_flow? && request.path !~ installation_routes_regex
+
+    redirect_to installation_new_path and return
```

Comment:
Aqui algo como:
```suggestion
    if Installation.installation_flow? && !request.path.include?('/installation')
      redirect_to installation_new_path and return
    end
```

Não fica mais simples de entender?

Reply:
Tem razão, vou testar para ver se n quebra os testes

Reply:
O autosave está mudando para esse formato:
```ruby
    return unless Installation.installation_flow? && !request.path.include?('/installation')

    redirect_to installation_new_path and return
``` 
Acho q dessa forma da para entender tbm, acho q o autosave n curte muito ifs 😄